processor/src/test/java/org/mapstruct/ap/test/bugs/_913/DomainDtoWithNvmsDefaultMapper.java

```diff
@@ -11,13 +11,17 @@
 import org.mapstruct.MappingTarget;
 import org.mapstruct.Mappings;
 import org.mapstruct.NullValueMappingStrategy;
+import org.mapstruct.NullValuePropertyMappingStrategy;
 import org.mapstruct.factory.Mappers;
 
 /**
  *
  * @author Sjaak Derksen
  */
-@Mapper( nullValueMappingStrategy = NullValueMappingStrategy.RETURN_DEFAULT, uses = Helper.class )
+@Mapper(
+    nullValueMappingStrategy = NullValueMappingStrategy.RETURN_DEFAULT, // This is wrong level.. should be on beanlevel
```

Comment:
this one worries me. Apparently the `NullValueMappingStrategy.RETURN_DEFAULT` has been (misused) to handle property mappings. To solve this I propose to have a `NullValuePropertyMappingStrategy.SET_TO_DEFAULT`. Note: this one also applies to non-update (regular) mappings to make it consistent.

Reply:
Yes you are right, it has been misused and it is actually my mistake 😄(I merged that PR). Using `NullValuePropertyMappingStrategy.SET_TO_DEFAULT` would make more sense here. The fact that it applies to regular mappings for collections is a mistake in my opinion. We should not be setting it then. The users should provide defaults in the `@ObjectFactory` or object constructor instead.

Reply:
I don't think we need to add anything for that. Honestly, I consider that behaviour as a bug and I am forced to do those mappings within `@AfterMapping`